.github/workflows/presubmit.yml

```diff
@@ -182,55 +87,32 @@ jobs:
       - uses: mpi4py/setup-mpi@v1
         with:
           mpi: intelmpi
-      - uses: actions/cache@v4
-        id: efficios_dep
-        env:
-          cache-name: cache-efficios_dep
-        with:
-          path: ~/efficios_dep/
-          key: ${{ runner.os }}-build-${{ env.cache-name }}
+      - uses: ./.github/actions/setup-thapi
+      - run: |
+          sudo apt install -y bats coreutils libpocl2 clinfo
       - uses: actions/download-artifact@v4
         with:
           name: thapi-bin
       - name: Untar THAPI
         run: tar -xvf thapi.tar
-      - run: sudo apt update; sudo apt install -y $APT_PACKAGE bats coreutils libpocl2 clinfo
-      - run: sudo gem install babeltrace2 opencl_ruby_ffi
-      - name: Load Efficios Dependencies
-        run: |
-          echo "$HOME/efficios_dep/bin" >> $GITHUB_PATH
-          echo "LD_LIBRARY_PATH=$HOME/efficios_dep/lib:$LD_LIBRARY_PATH" >> $GITHUB_ENV
       - name: Integration test
         run: |
           bats integration_tests/
 
   build-in-tree-and-check:
-    needs: [efficios_dep, pre_job]
+    needs: [pre_job]
     if: ${{ needs.pre_job.outputs.should_skip != 'true' }}
     name: Build in Tree ubuntu-24.04
     runs-on: ubuntu-24.04
     steps:
       - uses: actions/checkout@v4
-      - uses: actions/cache@v4
-        id: efficios_dep
-        env:
-          cache-name: cache-efficios_dep
-        with:
-          path: ~/efficios_dep/
-          key: ${{ runner.os }}-build-${{ env.cache-name }}
-      - run: sudo apt update; sudo apt install -y $APT_PACKAGE
-      - run: sudo gem install cast-to-yaml nokogiri babeltrace2 opencl_ruby_ffi metababel
-      - name: Load Efficios Dependencies
+      - uses: ./.github/actions/setup-thapi
+      - name: Make check
         run: |
-          echo "$HOME/efficios_dep/bin" >> $GITHUB_PATH
-          echo "PKG_CONFIG_PATH=$HOME/efficios_dep/lib/pkgconfig:$PKG_CONFIG_PATH" >> $GITHUB_ENV
-          echo "CPATH=$HOME/efficios_dep/include:$CPATH" >> $GITHUB_ENV
-          echo "LD_LIBRARY_PATH=$HOME/efficios_dep/lib:$LD_LIBRARY_PATH" >> $GITHUB_ENV
-          echo "LIBRARY_PATH=$HOME/efficios_dep/lib:$LIBRARY_PATH" >> $GITHUB_ENV
-      - run: ./autogen.sh
-      - run: ./configure
-      - run: make -j
-      - run: make check
+          ./autogen.sh
+          ./configure
+          make -j
+          make check
```

Comment:
I don't like these, this goes contradictory to the philosophy of github actions and increasing the diff for no good reason. I think that if we want to do that we should do so in a different PR.